index.js

```diff
@@ -115,21 +115,22 @@ TempMailbox.prototype.getMailDetail = function(id) {
 	return this.init_promise.then(function() { return defer.promise; });
 };
 
-TempMailbox.prototype.waitForEmail = function(summaryFilter, iterationDelay) {
+TempMailbox.prototype.waitForEmail = function(summaryFilter, iterationDelay, retries) {
 	var me = this;
 	
 	iterationDelay = iterationDelay || 2500;
+  retries = retries || -1;
 	
 	function checkLoop(mail) {
 		if (mail) {
 			var match = summaryFilter(mail);
 			if (match) {
 				return me.getMailDetail(mail.mail_id);
 			}
-			return me.getNextMailSummary().then(checkLoop);
+			return retries-- === 0 ? false : me.getNextMailSummary().then(checkLoop);
```

Comment:
This 'tight loop' path (without the `Promise.delay`) is used to quickly go through any mail in the queue.  If there are 10 emails in the queue, but say `retry == 3` is specified, then this would just return `false`.  If the intention of `retries` is to limit the total amount of time the function waits, then this isn't intended I think?

Reply:
Hi!
Most likely you are right.

I used the library in automated testing user registration in our project and found if something bad happened on our side and email wasn't send checkLoop never ends. 

So, I quickly added such workaround and it worked for us.

Unfortunately, guerillamail service API had been gone for a few days, so I moved to another library.

Reply:
Oh, humm - you mean the API is down?  The official site looks like it should still be there?

Reply:
Yes, exactly. The API was broken for several days but the site worked